tests/CsvReadTest.php

```diff
@@ -479,4 +491,50 @@ public function testInvalidFile()
         self::expectExceptionMessage('Invalid file: array');
         new CsvReader(['bad']);
     }
+
+    /**
+     * @dataProvider getPerformanceTestInputs
+     * @param string $fileContent
+     * @param int $expectedRows
+     * @param float $maxDuration
+     */
+    public function testPerformance($fileContent, $expectedRows, $maxDuration)
+    {
+        try {
+            $fileName = sys_get_temp_dir() . DIRECTORY_SEPARATOR . uniqid('perf-test');
+            file_put_contents($fileName, $fileContent);
+            $startTime = microtime(true);
+            $reader = new CsvReader($fileName);
+            $rows = 0;
+            foreach ($reader as $line){
+                $rows++;
+            }
+            $duration = microtime(true) - $startTime;
+            self::assertSame($expectedRows, $rows);
+            self::assertLessThanOrEqual($maxDuration, $duration);
+        } finally {
+            @unlink($fileName);
+        }
+    }
+
+    public function getPerformanceTestInputs()
```

Comment:
Pridal som perf testy. 

Podla nich na mojom PC trva spracovanie regulerneho vyrazu **`190ms`**, ktory sa pridaval, pri 10000 vzorke, ktora je nastavena v kode.

Napr. pre prvy data set `1M-simple-rows`, bola `duration`:
- v starom kode `4.0193018913269`
- v novom kode `4.1560678482056`
- teda rozdiel je do nejakych `5%`, pri opakovanych spusteniach to trocha kolisalo.

Reply:
ja bych rek, ze dobry - melo by to byt zhorseni o konstantu, takze v pohode